### PR TITLE
Fix generation of zuul projects.yaml

### PR DESCRIPTION
**Pull request by adrianfusco**
As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running

### CODE REVIEW

**Review by raukadah — COMMENTED** [review]
@adrianfusco Thank you for adding the improvements. Can we run `make run_molecule` command and also include zuul.d{projects.yaml, molecule.yaml} changes made by the script in this pr?

It will avoid unnecessary changes comes in a pr when new role gets added.

**Comment by pablintino** [comment]
There is still a "name" field at the end of the document that shouldn't be there. Example:
```
- project:
    github-check:
      jobs:
      - noop
      - ci-framework-crc-podified-galera-deployment
      - ci-framework-crc-podified-edpm-baremetal
      - ci-framework-crc-podified-edpm-deployment
      - cifmw-dev-prepare
      - cifmw-doc
      - cifmw-end-to-end
      - cifmw-end-to-end-nobuild-tagged
      - cifmw-kuttl
      - cifmw-edpm-build-images
      - cifmw-content-provider-build-images
      - cifmw-molecule-artifacts
      - cifmw-molecule-build_openstack_packages
      - cifmw-molecule-ci_setup
      - cifmw-molecule-copy_container
      - cifmw-molecule-discover_latest_image
      - cifmw-molecule-edpm_deploy
      - cifmw-molecule-edpm_prepare
      - cifmw-molecule-install_ca
      - cifmw-molecule-install_yamls
      - cifmw-molecule-libvirt_manager
      - cifmw-molecule-local_env_vm
      - cifmw-molecule-operator_build
      - cifmw-molecule-operator_deploy
      - cifmw-molecule-pkg_build
      - cifmw-molecule-registry_deploy
      - cifmw-molecule-repo_setup
      - cifmw-molecule-rhol_crc
      - cifmw-molecule-run_hook
      - cifmw-molecule-test_deps
      - cifmw-molecule-openshift_setup
      - cifmw-molecule-openshift_login
      - cifmw-molecule-edpm_deploy_baremetal
      - cifmw-molecule-edpm_build_images
      - cifmw-molecule-hive
      - cifmw-molecule-openshift_provisioner_node
      - cifmw-molecule-os_net_setup
      - cifmw-molecule-polarion
      - cifmw-molecule-set_openstack_containers
      - cifmw-molecule-tempest
      - cifmw-molecule-os_must_gather
      - cifmw-molecule-edpm_kustomize
      - cifmw-molecule-edpm_cr_overrides
    name: openstack-k8s-operators/ci-framework

````

**Comment by adrianfusco** [comment]
> There is still a "name" field at the end of the document that shouldn't be there. Example:
> 
> ```
> - project:
>     github-check:
>       jobs:
>       - noop
>       - ci-framework-crc-podified-galera-deployment
>       - ci-framework-crc-podified-edpm-baremetal
>       - ci-framework-crc-podified-edpm-deployment
>       - cifmw-dev-prepare
>       - cifmw-doc
>       - cifmw-end-to-end
>       - cifmw-end-to-end-nobuild-tagged
>       - cifmw-kuttl
>       - cifmw-edpm-build-images
>       - cifmw-content-provider-build-images
>       - cifmw-molecule-artifacts
>       - cifmw-molecule-build_openstack_packages
>       - cifmw-molecule-ci_setup
>       - cifmw-molecule-copy_container
>       - cifmw-molecule-discover_latest_image
>       - cifmw-molecule-edpm_deploy
>       - cifmw-molecule-edpm_prepare
>       - cifmw-molecule-install_ca
>       - cifmw-molecule-install_yamls
>       - cifmw-molecule-libvirt_manager
>       - cifmw-molecule-local_env_vm
>       - cifmw-molecule-operator_build
>       - cifmw-molecule-operator_deploy
>       - cifmw-molecule-pkg_build
>       - cifmw-molecule-registry_deploy
>       - cifmw-molecule-repo_setup
>       - cifmw-molecule-rhol_crc
>       - cifmw-molecule-run_hook
>       - cifmw-molecule-test_deps
>       - cifmw-molecule-openshift_setup
>       - cifmw-molecule-openshift_login
>       - cifmw-molecule-edpm_deploy_baremetal
>       - cifmw-molecule-edpm_build_images
>       - cifmw-molecule-hive
>       - cifmw-molecule-openshift_provisioner_node
>       - cifmw-molecule-os_net_setup
>       - cifmw-molecule-polarion
>       - cifmw-molecule-set_openstack_containers
>       - cifmw-molecule-tempest
>       - cifmw-molecule-os_must_gather
>       - cifmw-molecule-edpm_kustomize
>       - cifmw-molecule-edpm_cr_overrides
>     name: openstack-k8s-operators/ci-framework
> ```

@pablintino this key exists in the template: https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci/templates/projects.yaml#L6 isn't the same? It won't preserve the order since it's handled by yaml module.

**Comment by adrianfusco** [comment]
> @adrianfusco Thank you for adding the improvements. Can we run `make run_molecule` command and also include zuul.d{projects.yaml, molecule.yaml} changes made by the script in this pr?
> 
> It will avoid unnecessary changes comes in a pr when new role gets added.

Thanks @raukadah I've added the regenerated files.

File: zuul.d/molecule.yaml [hunk 1]
```diff
@@ -1,305 +1,304 @@
...
+    vars:
+      TEST_RUN: build_openstack_packages
 - job:
-    name: cifmw-molecule-test_deps
+    files:
+    - ^ansible-requirements.txt
+    - ^molecule-requirements.txt
+    - ^ci_framework/roles/operator_deploy/(?!meta|README).*
+    - ^ci/playbooks/molecule.*
+    name: cifmw-molecule-operator_deploy
```
Comment (cjeanner) [inline zuul.d/molecule.yaml:291]: it seems to lack the right nodeset - did you ensure you imported the current state?

File: zuul.d/molecule.yaml [hunk 2]
```diff
@@ -1,305 +1,304 @@
...
-      - ^ci/playbooks/molecule.*
+      TEST_RUN: polarion
 - job:
-    name: cifmw-molecule-ci_setup
+    files:
+    - ^ansible-requirements.txt
+    - ^molecule-requirements.txt
+    - ^ci_framework/roles/os_must_gather/(?!meta|README).*
+    - ^ci/playbooks/molecule.*
+    name: cifmw-molecule-os_must_gather
```
Comment (cjeanner) [inline zuul.d/molecule.yaml:17]: seems to lack the correct nodeset

File: zuul.d/molecule.yaml [hunk 3]
```diff
@@ -1,305 +1,304 @@
...
-      - ^ci_framework/roles/copy_container/(?!meta|README).*
-      - ^ci/playbooks/molecule.*
 - job:
-    name: cifmw-molecule-discover_latest_image
+    files:
+    - ^ansible-requirements.txt
+    - ^molecule-requirements.txt
+    - ^ci_framework/roles/set_openstack_containers/(?!meta|README).*
+    - ^ci/playbooks/molecule.*
+    name: cifmw-molecule-set_openstack_containers
```
Comment (cjeanner) [inline zuul.d/molecule.yaml:37]: pretty sure that one also needs to get a crc-based nodeset, when reading the reported CI failure.

**Review by cjeanner — CHANGES_REQUESTED** [review]
Beware, 3 jobs are apparently using the wrong nodeset - please double-check the generated content, and add the custom things accordingly.

File: zuul.d/molecule.yaml (hunk 2 above)
Comment (adrianfusco) [reply zuul.d/molecule.yaml:17]: Updated the parent based on @raukadah info https://github.com/openstack-k8s-operators/ci-framework/pull/401/files#diff-e11625dd628d0cfeeaad59207c1fabdd877fb86c87eeca532b83571e184628dcR20

File: zuul.d/molecule.yaml (hunk 3 above)
Comment (adrianfusco) [reply zuul.d/molecule.yaml:37]: Updated the parent based on @raukadah info https://github.com/openstack-k8s-operators/ci-framework/pull/401/files#diff-e11625dd628d0cfeeaad59207c1fabdd877fb86c87eeca532b83571e184628dcR17

File: zuul.d/molecule.yaml (hunk 1 above)
Comment (adrianfusco) [reply zuul.d/molecule.yaml:291]: Added nodeset https://github.com/openstack-k8s-operators/ci-framework/pull/401/files#diff-e11625dd628d0cfeeaad59207c1fabdd877fb86c87eeca532b83571e184628dcR15

**Review by cjeanner — COMMENTED** [review]
/approve

**Review by raukadah — APPROVED** [review]

**Pull request merged**